app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,63 @@
+class PasswordResetsController < ApplicationController
+  before_action :load_user, only: %i(edit update)
+  before_action :valid_user, only: %i(edit update)
+  before_action :check_expiration, only: %i(edit update)
+  def new; end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].downcase
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t "notifi.info_email"
+      redirect_to root_url
+    else
+      flash.now[:danger] = t "notifi.danger_email"
+      render :new
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add :password, t("notifi.update_error")
+      render :edit
+    elsif @user.update user_params
+      log_in @user
+      @user.update :reset_digest, nil
+      flash[:success] = t "notifi.update_success"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit :password, :password_confirmation
+  end
+
+  def load_user
+    @user = User.find_by email: params[:email]
+    return if @user
+    flash[:danger] = t "notifi.show"
+    redirect_to root_url and return unless true
```

Comment:
có cần `and return unless true` nữa không nhỉ? vì nếu sai nó đã return ngay dòng 45 rồi mà?